### PR TITLE
Add constraints to upgrade_metadata

### LINKED ISSUE #1005: Add constraint conversion to upgrade_metadata
### Problem Description

_As a user, it would be useful if the metadata upgraded my constraints as well._

### Expected behavior

- The `upgrade_metadata` method should also upgrade all constraints to match the format of the new metadata

### PR DESCRIPTION
**Pull request by amontanez24**
resolves #1005

### CODE REVIEW

File: tests/unit/metadata/test_metadata_upgrader.py [hunk 1]
```diff
@@ -0,0 +1,763 @@
...
+        {
+            'constraint_name': 'Positive',
+            'column_name': 'b'
+        },
+        {
+            'constraint_name': 'Positive',
+            'column_name': 'c'
+        },
+        {
+            'constraint_name': 'ScalarInequality',
```
Comment (fealho) [inline tests/unit/metadata/test_metadata_upgrader.py:426]: Why can't we pass the `strict` parameter to the Positive/Negative constraints in the new metadata?

File: tests/unit/metadata/test_metadata_upgrader.py [hunk 2]
```diff
@@ -0,0 +1,763 @@
...
+        },
+    ]
+    assert len(expected_constraints) == len(new_constraints)
+    for constraint in expected_constraints:
+        assert constraint in new_constraints
+
+
+@patch('sdv.metadata.metadata_upgrader.warnings')
+def test__upgrade_constraints_constraint_has_no_upgrade(warnings_mock):
+    """Test the ``_upfgrade_constraints`` method with constraints that can't be upgraded.
```
Comment (fealho) [inline tests/unit/metadata/test_metadata_upgrader.py:535]: typo, upgrade_constraints

File: tests/unit/metadata/test_metadata_upgrader.py [hunk 3]
```diff
@@ -0,0 +1,763 @@
...
+        }
+    ]
+    old_metadata = {'constraints': old_constraints}
+
+    # Run
+    new_constraints = _upgrade_constraints(old_metadata)
+
+    # Assert
+    assert new_constraints == []
+    warnings_mock.warn.assert_has_calls([
```
Comment (fealho) [inline tests/unit/metadata/test_metadata_upgrader.py:575]: Also assert there were only 4 calls.

**Review by fealho — CHANGES_REQUESTED** [review]
Good job! That was a long one to review, so it must've taken quite of bit of patience to actually code it 🎊 

I left a few minor comments, but after that it looks good to go.

**Review by fealho — APPROVED** [review]

**Review by pvk-developer — APPROVED** [review]

**Pull request merged**